src/AnonymousUser/AnonymousUserMiddleware.cs

```diff
@@ -26,6 +26,7 @@ public AnonymousUserMiddleware(RequestDelegate nextDelegate, AnonymousUserOption
         private async Task HandleRequestAsync(HttpContext httpContext)
         {
             var cookieEncoder = _options.EncoderService ?? throw new NullReferenceException($"{nameof(_options.EncoderService)} is null and should have a valid encoder.");
+            _ = _options.UserIdentifierFactory ?? throw new NullReferenceException($"{nameof(_options.UserIdentifierFactory)} is null and should have a valid factory.");
```

Comment:
```suggestion
            var cookieEncoder = _options.EncoderService ?? throw new ArgumentNullException(nameof(_options.EncoderService), $"{nameof(_options.EncoderService)} is null and should have a valid encoder.");
            _ = _options.UserIdentifierFactory ?? throw new ArgumentNullException(nameof(_options.EncoderService), $"{nameof(_options.UserIdentifierFactory)} is null and should have a valid factory.");
```

I'd use an `ArgumentNullException` here.